src/software/sensor_fusion/refbox_data.h

```diff
@@ -0,0 +1,259 @@
+    // Otherwise, times are ordered from smallest to largest.
+    std::vector<int> yellow_card_times;
+    // The total number of yellow cards ever issued to the team.
+    int yellow_cards;
+    // The number of timeouts this team can still call.
+    // If in a timeout right now, that timeout is excluded.
+    int timeouts;
+    // The number of microseconds of timeout this team can use.
+    int timeout_time;
+    // The pattern number of this team's goalkeeper.
+    int goalkeeper;
+    // The total number of countable fouls that act towards yellow cards
+    int foul_counter;
+    // The number of consecutive ball placement failures of this team
+    int ball_placement_failures;
+    // Indicate if the team is able and allowed to place the ball
+    bool can_place_ball;
+    // The maximum number of bots allowed on the field based on division and cards
+    int max_allowed_bots;
+};
+
+/**
+ * RefboxData is a container class to represent the referee proto internally
+ * It renames fields for ease of use, i.e. command->game_state and blue/yellow->us/them
+ * and uses our data structures, such as Timestamp, Duration, and Point
+ */
+class RefboxData
+{
+   public:
+    RefboxData(Timestamp packet_timestamp, Timestamp game_state_timestamp,
+               int game_state_counter, Point designated_position,
+               bool blue_team_on_positive_half,
+               Duration current_game_state_time_remaining, TeamInfo us_team_info,
+               TeamInfo them_team_info, RefboxGameState game_state,
+               RefboxGameState next_game_state, RefboxStage stage,
+               std::vector<GameEvent> game_events,
+               std::vector<ProposedGameEvent> proposed_game_events)
+        : packet_timestamp(packet_timestamp),
+          game_state_timestamp(game_state_timestamp),
+          game_state_counter(game_state_counter),
+          designated_position(designated_position),
+          blue_team_on_positive_half(blue_team_on_positive_half),
+          current_game_state_time_remaining(current_game_state_time_remaining),
+          us_team_info(us_team_info),
+          them_team_info(them_team_info),
+          game_state(game_state),
+          next_game_state(next_game_state),
+          stage(stage),
+          game_events(game_events),
+          proposed_game_events(proposed_game_events)
+    {
+    }
+
+    Timestamp getPacketTimestamp(void)
+    {
+        return packet_timestamp;
+    }
+
+    Timestamp getGameStateTimestamp(void)
+    {
+        return game_state_timestamp;
+    }
+
+    int getGameStateCounter(void)
+    {
+        return game_state_counter;
+    }
+
+    Point getDesignatedPosition(void)
+    {
+        return designated_position;
+    }
+
+    bool getBlueTeamOnPositiveHalf(void)
+    {
+        return blue_team_on_positive_half;
+    }
+
+    Duration getCurrentGameStateTimeRemaining(void)
+    {
+        return current_game_state_time_remaining;
+    }
+
+    TeamInfo getUsTeamInfo(void)
```

Comment:
should we just use `getFriendlyTeamInfo` for consistency even though proto says us/them?

Reply:
fixed